qiskit/qobj/qasm_qobj.py

```diff
@@ -0,0 +1,529 @@
+    def n_qubits(self, value):
+        self._data['n_qubits'] = value
+
+    def to_dict(self):
+        """Return a dictionary format representation of the QASM Qobj config.
+
+        Returns:
+            dict: The dictionary form of the QasmQobjConfig.
+        """
+        return self._data
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new QasmQobjConfig object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the config
+
+        Returns:
+            QasmQobjConfig: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __getattr__(self, name):
+        try:
+            return self._data[name]
+        except KeyError:
+            raise AttributeError('Attribute %s is not defined' % name)
+
+    def __eq__(self, other):
+        if isinstance(other, QasmQobjConfig):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class QobjDictField:
+    """A class used to represent a dictionary field in Qobj
+
+    Exists as a backwards compatibility shim around a dictionary for Qobjs
+    previously constructed using marshmallow.
+    """
+
+    _data = {}
+
+    def __init__(self, **kwargs):
+        """Instantiate a new Qobj dict field object.
+
+        Args:
+            kwargs: arbitrary keyword arguments that can be accessed as
+                attributes of the object.
+        """
+        self._data = kwargs
+
+    def __getstate__(self):
+        return self._data
+
+    def __setstate__(self, state):
+        self._data = state
+
+    def __getattr__(self, attr):
+        try:
+            return self._data[attr]
+        except KeyError:
+            raise AttributeError('Attribute %s is not defined' % attr)
+
+    def __setattr__(self, name, value):
+        if not hasattr(self, name):
+            self._data[name] = value
+        else:
+            super().__setattr__(name, value)
+
+    def to_dict(self):
+        """Return a dictionary format representation of the QASM Qobj.
+
+        Returns:
+            dict: The dictionary form of the QobjHeader.
+
+        """
+        return self._data
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new QobjHeader object from a dictionary.
+
+        Args:
+            data (dict): A dictionary representing the QobjHeader to create. It
+                will be in the same format as output by :func:`to_dict`.
+
+        Returns:
+            QobjHeader: The QobjHeader from the input dictionary.
+        """
+
+        return cls(**data)
+
+    def __eq__(self, other):
+        if isinstance(other, QobjDictField):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class QasmQobjExperimentHeader(QobjDictField):
+    """A header for a single QASM experiment in the qobj."""
+
+    def __eq__(self, other):
+        if isinstance(other, QasmQobjExperimentHeader):
```

Comment:
Can't the parent's (`QobjDictField`) `__eq__` be implemented as 
```
if type(self) == type(other) and self.do_dict() == other.to_dict()
```
so each child doesn't need to reimplement this?

Reply:
This is a good idea, fixed in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/13eed6505f5990f7bf966c0eb1d3951ae35d4e07